classes.py

```diff
@@ -0,0 +1,544 @@
+import json
+import math
+import os
+from urllib.parse import urlparse, parse_qs
+import csv
+
+import requests
+from bs4 import BeautifulSoup
+
+_CSV_DELIMITER = ';'
+_URLDOMAIN = "https://chess-results.com"
+_RATING_LIST_HEADER = 'Id_No;Id_CBX;Title;Name;Rtg_Nat;ClubName;Birthday;Sex;Fed;TotalNumGames;SumOpponRating' \
+                      ';TotalPoints'
+_MAX_NUM_GAMES_TEMP_RATING = 15
+_K_STARTING_NUM_GAMES = [(30, 0),  # grampo
+                         (25, _MAX_NUM_GAMES_TEMP_RATING),  # 15
+                         (15, 40),
+                         (10, 80)]
+_CURRENT_TOURNAMENT_SNR = 0
+
+
+class FexerjRatingCycle:
+
+    def __init__(self, tournaments_file, first_item, items_to_process, initial_rating_filepath):
+
+        self.tournaments_file = tournaments_file
+        self.first_item = first_item
+        self.items_to_process = items_to_process
+        self.rating_list = {}
+        self.tournaments = {}
+        self.cbx_to_fexerj = {}
+        self.manual_entries = {}
+
+        self.load_manual_entry_dict()
+
+        with open(tournaments_file, 'r') as f:
+            reader = csv.reader(f, delimiter=_CSV_DELIMITER)
+            self.tournaments = list(reader)[1:]
+            for tournament in self.tournaments:
+
+                final_rating_filepath = "RatingList_after_%s.csv" % (tournament[0])
+
+                if int(tournament[0]) in range(first_item, first_item + items_to_process):
+
+                    print("\nRunning tournament %s (%s)...\n" % (tournament[0], tournament[2]))
+
+                    self.get_rating_list(initial_rating_filepath)
+
+                    print("Reading from %s" % initial_rating_filepath)
+                    print("Writing to %s" % final_rating_filepath)
+
+                    trn_type = tournament[4]
```

Comment:
(opt) Nowadays Python3 supports enums. Maybe we should cast the string to an enum here.